app/models/asset_group.rb

```diff
@@ -14,23 +14,26 @@ class AssetGroup < ActiveRecord::Base
   has_many :asset_group_assets
   has_many :assets, :through => :asset_group_assets
 
+  has_many :samples, :through => :assets
+
   validates :name, :presence => true, :uniqueness => true
   validates :study, :presence => true
 
  scope :for_search_query, ->(query,with_includes) { where([ 'name LIKE ?', "%#{query}%" ]) }
 
   def all_samples_have_accession_numbers?
-    unaccessioned_samples.count == 0
+    unaccessioned_samples.empty?
   end
 
+  # The has many through only works if the asset_group_assets are stored in the database,
+  # which won't be the case for new records. We depend on checking this on unsaved
+  # asset groups during the submission process. Here we switch between to scopes.
   def unaccessioned_samples
-    Sample.find(:all,
-      :joins => [
-        'INNER JOIN aliquots ON aliquots.sample_id = samples.id',
-        'INNER JOIN sample_metadata ON sample_metadata.sample_id = samples.id'
-        ],
-        :conditions => ['aliquots.receptacle_id IN (?) AND sample_ebi_accession_number IS NULL',assets.map(&:id)]
-    )
+    if new_record?
+      Sample.contained_in(assets).without_accession
```

Comment:
If asset_group_assets  are not stored, seems that assets and samples depend on it:
has_many :asset_group_assets
has_many :assets, :through => :asset_group_assets
has_many :samples, :through => :assets

What would happen in this case with Sample.contained_in(assets)?